src/Generator/Formatting.fs

```diff
@@ -0,0 +1,431 @@
+        | GLType.GLDEBUGPROCARB -> specTypeToCSharpTypeWithFallback "GLDEBUGPROCARB"
+        | GLType.GLDEBUGPROCKHR -> specTypeToCSharpTypeWithFallback "GLDEBUGPROCKHR"
+        | GLType.RefPointer ty ->
+             "ref " + typeToString ty
+        | GLType.OpenToolkit ty ->
+            match ty with
+            | Vector2 -> "Vector2"
+            | Vector3 -> "Vector3"
+            | Vector4 -> "Vector4"
+            | Matrix4 -> "Matrix4"
+    
+    let typeToStringCached ty =
+        match cache.TryGetValue ty with
+        | true, value -> value
+        | false, _ ->
+            let res = typeToString ty
+            cache.[ty] <- res
+            res
+
+let inline typeToString ty = TypeToString.typeToStringCached ty
+
+
+let (|StartsWith|_|) (pattern:string) (str:string) =
+    if str.StartsWith pattern then str.Substring(pattern.Length) |> Some
+    else None
+
+let (|EndsWith|_|) (pattern: string) (str: string) =
+    if str.EndsWith pattern then str.Substring(0, str.Length - pattern.Length) |> Some
+    else None
+
+let formatNameRemovingPrefix (name: string) =
+    let prefix =
+        prefixToRemove
+        |> Array.tryFind(fun prefix -> name.StartsWith prefix)
+    match prefix with
+    | Some prefix ->
+        let res = name.Substring prefix.Length
+        match res.[0] with
+        | '1' | '2' | '3' | '4' | '5'
+        | '6' | '7' | '8' | '9' ->
+            "_" + res
+        | _ -> res
+    | _ -> name
+
+let formatFunctionName (name: string) =
+    let nameWithRemovedPrefix =
+        formatNameRemovingPrefix name
+    let sufix =
+        sufixToRemove
+        |> Array.tryFind(fun sufix -> nameWithRemovedPrefix.EndsWith sufix)
+    match sufix with
+    | Some sufix -> nameWithRemovedPrefix.Substring(0, nameWithRemovedPrefix.Length - sufix.Length)
+    | _ -> nameWithRemovedPrefix
+
+let formatParameterName (name: string) =
+    reservedKeywords
+    |> Array.tryFindIndex (fun n -> name = n)
+    |> Option.map(fun i -> reservedKeywordsUpper |> Array.item i)
+    |> Option.defaultValue name
+
+let formatParam (p: TypedParameterInfo) = formatParameterName p.name
+
+let namespaceForGlSpecification (openGl: RawOpenGLSpecificationDetails) =
+    let prettyName =
+        let temp = openGl.version
+        let versionAsString =
+            let str = openGl.versionNumber |> string
+            str.Replace(".", "")
+        match temp with
+        | StartsWith "GL_ES_VERSION" _
+        | StartsWith "GL_VERSION_ES_CM_1_0" _ -> ".ES" + versionAsString
+        | StartsWith "GL_VERSION" _ -> versionAsString
+        | _ -> temp
+    prettyName
+    |> sprintf "GL%s"
+
+module PrintReady =
+
+    let formatEnumGroup (enumGroup: EnumGroup) =
+        let prettyEnumCases =
+            enumGroup.cases
+            |> Array.Parallel.map(fun case -> 
+                { actualName = case.name
+                  prettyName = case.name |> formatNameRemovingPrefix
+                  value = case.value } : PrintReadyEnum
```

Comment:
Why specify the printreadyenum type?

Reply:
Shared data along threads, the pipeline computers all prettified names in advance for the parallel call.

Same goes for all PrintReady types